### ISSUE TITLE
Running optimize rasters script

### ISSUE THREAD

**Issue by B-Carson** [Esri/OptimizeRasters#56]
I'm trying to run the Optimize Rasters script to generate a MRF which will reside in S3 from a directory of TIFF images (8-bit unsigned). However, I'm running into a warning which is causing the tool to fail 17 seconds into execution. The warning is as follows:

 

Executing: OptimizeRasters Imagery_to_MRF_LERC Local Profile " " C:\arcgis\ArcTutor\Raster\Data\Amberg_tif # "Amazon S3" pyadav brianmrftesting brianmrftesting/optimizerasters \\sertao\rasterdrive\MRF_for_AWS_s3\OutputTempFolder_OR \\sertao\rasterdrive\MRF_for_AWS_s3\RasterProxyOutputFolder \\sertao\rasterdrive\MRF_for_AWS_s3\CacheFolder # "Mode mrf;RasterFormatFilter tif,tiff,TIF,TIFF,img,jp2,JP2,IMG,JPG,jpg,sid,SID;ExcludeFilter tmp,rrd,idx,lrc,mrf_cache,pjp,ppng,pft,pzp,ovr,aux.xml,aux,tfw,TFW,pjg;IncludeSubdirectories true;Compression LERC;Quality #;LERCPrecision #;BuildPyramids true;PyramidFactor #;PyramidSampling avg;PyramidCompression LERC;NoDataValue #;BlockSize 512;Scale #;KeepExtension false;Threads 4;GDAL_Translate_UserParameters #"

Start Time: Wed May 30 11:25:44 2018

Running script OptimizeRasters...

log-msg:Using output endpoint> https://brianmrftesting.s3-us-west-1.amazonaws.com/

log-critical:maximum recursion depth exceeded while calling a Python object

log-critical:Unable to initialize the (S3) upload module! Check module setup/credentials. Quitting..

log-status:[Failed!]

Completed script OptimizeRasters...

Succeeded at Wed May 30 11:25:59 2018 (Elapsed Time: 15.86 seconds)

 

I have checked to ensure that my setup/credentials are correct, but I'm unable to get past the error message.

**Comment by Chamlika** [comment]
@B-Carson Could you check if there's an OptimizeRasters job file with the extension .orjob file created where the Optimizerasters is installed/where the Optimizerasters.py is located. If there's a job file created each time you try to run this command, could you tell me how big in size that job file is. If there isn't such a job file created, it's possible the depth of subfolders to process at the local source location is more than expected. Could you also see if giving a single input folder that do not have that many subfolders internally solves the issue?

**Comment by B-Carson** [comment]
@Chamlika 

It ended up being a bad connection as a result of the settings in the Profile Editor. Once I reran the Profile Editor with the corrected parameters, I was able to run my source TIFF files through the OptimizeRasters script.

However, I'm noticing some odd behavior in 10.5.1, 10.6 and ArcGIS Pro 2.2. In 10.5.1 and 10.6, I can bring in a single raster proxy file which was created from OptimizeRasters, but it will not display. Zooming to the layer does take us to the correct geographic location though. In ArcGIS Pro 2.2, after adding the raster proxy for the MRF to the map, we see a solid black square which represents the extent; again in the correct location.

I was able to add the raster proxy files to a mosaic dataset, and 11 of 12 images display as expected. However, as the raster proxy files are recognized as raster datasets, shouldn't we be able to add them as individual files in both ArcMap and ArcGIS Pro?

Thanks,

Brian
